### PR TITLE
Add scalafmt

### PR DESCRIPTION
**Pull request by mdedetrich**
Now that scalafmt 3.0.0 has come out (which works with scala3 dialect without problems) we can finally use it on protoquill! Some notes

1. We use a fairly standard `.scalafmt.conf` configuration, I have personal preferences  that are different (i.e. I can suggest 
https://github.com/mdedetrich/akka-streams-json/blob/master/.scalafmt.conf), let me know if you want to try something different. The PR is comprised of 2 commits so its easy to rebase and update it
2. A github workflow has been added that uses scalafmt-native to independently (and hence concurrently) check that the codebase is formatted on every PR. Since its native the check is very fast (i.e. 5-10 seconds)

### CODE REVIEW

**Comment by mdedetrich** [comment]
Because of https://github.com/scalameta/scalafmt/issues/2687 I am converting this PR to a draft, once this gets fixed I will retry formatting

**Comment by deusaquilus** [comment]
Can we make `maxColumn=180` and see what happens?

**Comment by mdedetrich** [comment]
@deusaquilus I just forced pushed with `maxColumn=180`

**Comment by mdedetrich** [comment]
@deusaquilus I have just rebased off the latest master

File: .scalafmt.conf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+runner.dialect = scala3
+version=3.0.1
+maxColumn=180
+project.git = true
+align.preset = more
```
Comment (deusaquilus) [inline .scalafmt.conf:5]: Let's go back to `some` for this one. I'm not comfortable with all the alignments of assignments that are happening.

Comment (mdedetrich) [reply .scalafmt.conf]: Done rebased

**Comment by deusaquilus** [comment]
Could we skip formatting build.sbt? I don't feel very comfortable with that.

**Comment by mdedetrich** [comment]
I will have a look into it but there shouldn't be any problems since its Scala code like anything else, is there something specific that you are concerned about?

There are projects which have build definitions equally (if not more) complicated that use scalafmt

**Comment by deusaquilus** [comment]
It's not a system thing. It's just that I experiment with build.sbt with one-line changes that I don't want other tools to introduce other things that could confuse me. SBT syntax already annoys me so, so much that my head might explode if there was more stuff on top of it.

**Comment by mdedetrich** [comment]
Sure thing, I think I found a way to do it

**Comment by deusaquilus** [comment]
Also, let's please only do scalafmt for quill-sql in protoquill for now. I'm about to make things like JdbcContextBase, ZioJdbcContext etc... 100% identical to Scala2-Quill which is how I hope to sanely manage two codebases as the code evolves. If I can't diff JdbcContextBase etc... with the Scala2-Quill equivalents, my head also will probably eventually explode.

**Comment by mdedetrich** [comment]
Does it just make sense to leave it after your done with the merging.

We can do both the scala2 and proto at the same time with the exact same scalafmt which means the diffs will only show relevant things (along with scala2 to scala3 syntax change)

**Comment by deusaquilus** [comment]
Yeah, we should probably do Scalafmt for quill-jdbc and quill-jdbc-zio etc.... in Scala2-Quill but I really want to do that in a separate step. I want to do it to be know in the git-history that JdbcContextBase, ZioJdbcContext etc... were identical in Quill/ProtoQuill _before_ formatting was applied. Then there should be a "add scalafmt to quill-jdbc, quill-jdbc-zio" commit to both repos so lineage is completely clean. Otherwise again.... :exploding_head:

**Comment by deusaquilus** [comment]
(I have some mental scars from codebases that were supposed to be exactly the same that diverged due to different formatting standards introduced into one of them... and it took weeks to bring then back together, for totally no reason. So I want to do this stuff really carefully.)

**Comment by mdedetrich** [comment]
Okay I think it just makes sense to do this after you have done the migration and verified everything is okay, there isn't any point in unnecessarily stressing you out.

One thing that we may want to look into however is if there is tool that automatically migrates scala2 syntax to scala3, scalafix might be able to do this and if so are you interested in it?

**Comment by deusaquilus** [comment]
Yeah, definitely. Learning Scalafix is on my TODO.

File: build.sbt [hunk 2]
```diff
@@ -13,7 +12,7 @@ ThisBuild / versionScheme := Some("always")
 // https://github.com/sbt/sbt-pgp/issues/178
 Global / useGpgPinentry := true
```
[the remark this answers is not in the document]
Comment (i10416) [reply build.sbt:14]: Looking at the thread and I found you do not do auto-format Intentionally? If so, feel free to ignore this suggestion.

**Comment by guizmaii** [comment]
Closing as it's using scalafmt now

**Pull request closed without merging**